### DISCUSSION TITLE
Widget icon not displayed correctly

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#3642] [Q&A] [answered]
So what I've done:

- I've "created" a widget class that inherits from `OpenWeather` widget
```
from urllib.error import URLError
from libqtile.log_utils import logger
from qtile_extras.widget import OpenWeather
from libqtile.widget.open_weather import OpenWeatherResponseError, _OpenWeatherResponseParser
from libqtile.widget import base

colors = [
    "#1E1E2E",  # 0
    "#302D41",  # 1
    "#575268",  # 2
    "#988BA2",  # 3
    "#F5E0DC",  # 4
    "#C9CBFF",  # 5
    "#D9E0EE",  # 6
    "#B5E8E0",  # 7
    "#89DCEB",  # 8
    "#96CDFB",  # 9
    "#81a1c1",  # 10
    "#F28FAD",  # 11
    "#F8BD96",  # 12
    "#FAE3B0",  # 13
    "#ABE9B3",  # 14
    "#DDB6F2",  # 15
]

class MyWeather(OpenWeather):

    defaults = [
        ("appkey", "ce4579dd88a8d4877a8c23f2a10d61cc", ""),
        ("padding", 10, ""),
        ("fontsize", 30, ""),
        ("foreground", "#D9E0EE", ""),
        ("format", "{main_feels_like:.0f}°{units_temperature} {icon}", ""),
    ]

    def __init__(self, **config):
        OpenWeather.__init__(self, **config)
        self.add_defaults(MyWeather.defaults)

    def parse(self, response):
        try:
            rp = _OpenWeatherResponseParser(response, self.dateformat, self.timeformat)
        except OpenWeatherResponseError as e:
            return "Error {}".format(e.resp_code)

        data = rp.data
        data["units_temperature"] = "C" if self.metric else "F"
        data["units_wind_speed"] = "Km/h" if self.metric else "m/h"
        data["icon"] = self.symbols.get(data["weather_0_icon"], self.symbols["Unknown"])
        if data["temp"] >= 15:
            self.foreground = colors[13]
        else:
            self.foreground = colors[9]
        return self.format.format(**data)
```
widget instance(s)
```
        widget.MyWeather(
            appkey="ce4579dd88a8d4877a8c23f2a10d61cc",
            foreground=colors[6],
            padding=10,
            font="CodeNewRoman Nerd Font Mono Bold",
            fontsize=37,
            format="{main_feels_like:.0f}°{units_temperature} {icon}",
            location=location(),
            mouse_callbacks={"Button1": weather_popup()},
            **decor
        ),
```
(weather_popup is just a function that does a qtile_extras popup, location is a... location function :) )
- I've used it on my last machine without any issues
- Having switched to my current machine, this issue appears.

Comparison between the two widgets on the different machines (the bottom one is from VNC viewer, so it's a bit blurry, sorry)
![myweather-comparison](https://user-images.githubusercontent.com/84532402/174513306-9a9f9415-eeca-40ca-85e8-e9f9fe1032e7.png) this is my former machine

Both have the same implementation, both have the same way of being called(sorry for butchering English, but you get the point)...

I just don't get the difference in icons.

Last but not least, my [configuration](https://github.com/ervinpopescu/dots/tree/main/.config/qtile)

LE:
![image](https://user-images.githubusercontent.com/84532402/174516491-48707ad1-522e-4d39-962e-e04371913965.png)
Getting closer... Just made sure I had the same fonts on both of the machines

**Comment by ervinpopescu** [comment, marked as answer]
Solved... Just turned hinting on and off in lxappeareance... Have no idea if that did it tho

**Reply by elParaguayo** [reply]
You may want to delete your app key from the above code...

**Reply by ervinpopescu** [reply]
> You may want to delete your app key from the above code...

Well it's one from the docs, I think...

**Reply by elParaguayo** [reply]
Oh, ok. Assumed it was yours!
